### PR TITLE
online-judge-verify-helper: init at 5.6.0

### PR DESCRIPTION
**Pull request by sei40kr**
###### Motivation for this change

Add [online-judge-verify-helper](https://github.com/online-judge-tools/verification-helper).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/online-judge-verify-helper/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+buildPythonApplication rec {
+  pname = "online-judge-verify-helper";
+  version = "5.6.0";
+
+  src = fetchFromGitHub {
+    owner = "online-judge-tools";
+    repo = "verification-helper";
+    rev = "v${version}";
+    sha256 = "sha256-sBR9/rf8vpDRbRD8HO2VNmxVckXPmPjUih7ogLRFaW8=";
```
Comment (urandom2) [inline pkgs/tools/misc/online-judge-verify-helper/default.nix:21]: nit
```suggestion
    hash = "sha256-sBR9/rf8vpDRbRD8HO2VNmxVckXPmPjUih7ogLRFaW8=";
```

File: pkgs/tools/misc/online-judge-verify-helper/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  patches = [ ./fix-paths.patch ];
+  postPatch = ''
+    substituteInPlace onlinejudge_verify/main.py --subst-var-by git ${git}
```
Comment (urandom2) [inline pkgs/tools/misc/online-judge-verify-helper/default.nix:26]: did you mean?
```suggestion
    substituteInPlace onlinejudge_verify/main.py --subst-var-by git ${git}/bin/git
```

File: pkgs/tools/misc/online-judge-verify-helper/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    pyyaml
+    setuptools
+    toml
+  ];
+
+  # Needs internet to run tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "a testing framework for snippet libraries used in competitive programming";
```
Comment (urandom2) [inline pkgs/tools/misc/online-judge-verify-helper/default.nix:42]: nit
```suggestion
    description = "A testing framework for snippet libraries used in competitive programming";
```

File: pkgs/tools/misc/online-judge-verify-helper/default.nix (hunk 2 above)
Comment (sei40kr) [reply pkgs/tools/misc/online-judge-verify-helper/default.nix:26]: `/bin/git` is appended here:
https://github.com/NixOS/nixpkgs/pull/157294/files#diff-4fcb2c4716deaeefd540e70d71ad27b802daf485c5cd20250f4a808bab1f5673R11-R12

File: pkgs/tools/misc/online-judge-verify-helper/default.nix (hunk 3 above)
Comment (sei40kr) [reply pkgs/tools/misc/online-judge-verify-helper/default.nix]: @urandom2 
Thanks! Fixed.

File: pkgs/tools/misc/online-judge-verify-helper/default.nix (hunk 1 above)
Comment (sei40kr) [reply pkgs/tools/misc/online-judge-verify-helper/default.nix]: @urandom2 Thanks! Fixed.

**Review by urandom2 — APPROVED** [review]

File: pkgs/tools/misc/online-judge-verify-helper/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/tools/misc/online-judge-verify-helper/default.nix:26]: Why not wrap the binary and add git to PATH instead? that is less patching and easier to update

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -5155,6 +5155,8 @@ with pkgs;
 
   online-judge-tools = with python3.pkgs; toPythonApplication online-judge-tools;
 
+  online-judge-verify-helper = python3.pkgs.callPackage ../tools/misc/online-judge-verify-helper { };
+
```
Comment (getchoo) [inline pkgs/top-level/all-packages.nix:5159]:
```suggestion
```
this can go in pkgs/by-name nowadays

**Review by getchoo — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Closing as stale

**Pull request closed without merging**